### ISSUE TITLE
net.set_eval() seems not work well

### ISSUE THREAD

**Issue by qzhiyue** [tensorlayer/TensorLayerX#59]
### Issue Description
When I test my pspnet model, I find if not use "with torch.no_grad()" or "gradient()", the gpu memory will be full after testing several photos. I guess set_eval() function seems to have failed. Or I used the wrong method to test? This is my code, thank you!

In addition, I found that the batch size will affect the final test results. If net. eval() is not performed in the pytorch, it will cause similar problems. It seems that this is caused by the BatchNorm layer.

```python
    os.environ['TL_BACKEND'] = 'torch'
    tlx.set_device(device='GPU', id=3)
    # ...
    net = models[backend]()
    net.load_weights('test.npz', format='npz_dict', skip=True)
    test_dataset = MyDataset(root_dir="test/")
    test_loader = DataLoader(test_dataset, batch_size=4, shuffle=True)

    train_weights = net.trainable_weights
    scheduler = tlx.optimizers.lr.StepDecay(learning_rate=0, step_size=30, gamma=0.5, last_epoch=-1)
    optimizer = tlx.optimizers.Adam(lr=scheduler)

    hist = np.zeros((num_classes, num_classes))
    net.set_eval()
    # with torch.no_grad():
    for x, y, y_cls in test_loader:
        _out, _out_cls = net(x)
        seg_loss = tlx.losses.softmax_cross_entropy_with_logits(_out, y)
        cls_loss = tlx.losses.sigmoid_cross_entropy(_out_cls, y_cls)
        _loss = seg_loss + 1 * cls_loss
        # grads = optimizer.gradient(_loss, train_weights)
        # optimizer.apply_gradients(zip(grads, train_weights))
        '''
            compute miou matrix
        '''
        out = tlx.convert_to_numpy(_out)
        y = tlx.convert_to_numpy(y)
        out = np.argmax(out, axis=1)
        for i in range(0, out.shape[0]):
            pred = out[i]
            gt = y[i]
            hist += fast_hist(gt.flatten(), pred.flatten(), num_classes)
            
    # compute miou then print
    mIoUs = per_class_iu(hist)
    for ind_class in range(num_classes):
        print('===>' + name_classes[ind_class] + ':\t' + str(round(mIoUs[ind_class] * 100, 2)))
    print('===> mIoU: ' + str(round(np.nanmean(mIoUs) * 100, 2)))
    print("test loss: {}".format(train_loss))

```

**Comment by Laicheng0830** [comment]
Resolved, please see [here](https://github.com/tensorlayer/TensorLayerX/commit/be76ccf3a984214e1f7e6759b7c07520f84a1b1f)
